src/lib/notificationUtils.js

```diff
@@ -0,0 +1,32 @@
+'use strict'
+
+const request = require('co-request')
+const _ = require('lodash')
+const url = require('url')
+
+function isHTTPS (uri) {
+  return url.parse(uri).protocol.match(/https/) !== null
+}
+
+function tlsOptions (target, config) {
+  const tls = config.get('tls')
+  const useTLS = isHTTPS(target) && tls
+  return useTLS ? _.omit({
```

Comment:
Why `omit`? Do we expect any of these values to be undefined if `tls` is defined?